app/client/rest/posts.ts

```diff
@@ -194,13 +194,10 @@ const ClientPosts = (superclass: any) => class extends superclass {
         );
     };
 
-    searchPostsWithParams = async (teamId: string, params: any) => {
-        this.analytics.trackAPI('api_posts_search', {team_id: teamId});
-
-        return this.doFetch(
-            `${this.getTeamRoute(teamId)}/posts/search`,
-            {method: 'post', body: params},
-        );
+    searchPostsWithParams = async (teamId: string, params: PostSearchParams) => {
+        this.analytics.trackAPI('api_posts_search');
+        const endpoint = teamId === '' ? `${this.getPostsRoute()}/search` : `${this.getTeamRoute(teamId)}/posts/search`;
```

Comment:
Nit: we can check just `teamId` instead of `teamId === ''`, right? Not sure which one is better thought.